File: day1/java/perfectsqr.java
import java.util.Scanner;
class PerfectSquareFactors {
    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {
        
            System.out.print("Enter the lower bound of the range: ");
            int lowerBound = scanner.nextInt();
        
            System.out.print("Enter the upper bound of the range: ");
            int upperBound = scanner.nextInt();
        
            System.out.println("Numbers with at least one perfect square factor in the range:");
            for (int i = lowerBound; i <= upperBound; i++) {
                if (hasPerfectSquareFactor(i)) {
                    System.out.println(i);
                }
            }
        }
        catch (Exception e) {
            System.out.println("Error occurred: " + e.getMessage());
        }
        finally {
            System.out.println("Exiting program.");
        }
    }
    
    private static boolean hasPerfectSquareFactor(int number) {
        for (int i = 2; i * i <= number; i++) {
            if (number % (i * i) == 0) {
                System.out.print(i*i+"->");
                return true;
            }
        }
        return false;
    }
}

File: day1/Python/perfectsqr.py
import math

def has_perfect_square_factor(n):
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % (i * i) == 0:
            print(i*i,end="->")
            return True
    return False

def perfect_square_factors(start, end):
    for num in range(start, end + 1):
        if has_perfect_square_factor(num):
            print(num)

# Example usage
start = int(input("Enter start number: "))
end =  int(input("Enter end number: "))
perfect_square_factors(start, end)